app/src/main/java/ch/epfl/sdp/game/Solo.java

```diff
@@ -0,0 +1,158 @@
+package ch.epfl.sdp.game;
+
+import android.util.Log;
+
+import java.util.ArrayList;
+
+import ch.epfl.sdp.artificial_intelligence.RandomEnemyGenerator;
+import ch.epfl.sdp.artificial_intelligence.SinusoidalMovement;
+import ch.epfl.sdp.database.firebase.api.SoloDatabaseAPI;
+import ch.epfl.sdp.entity.Enemy;
+import ch.epfl.sdp.entity.Player;
+import ch.epfl.sdp.entity.PlayerManager;
+import ch.epfl.sdp.geometry.Area;
+import ch.epfl.sdp.geometry.CircleArea;
+import ch.epfl.sdp.geometry.GeoPoint;
+import ch.epfl.sdp.geometry.UnboundedArea;
+import ch.epfl.sdp.item.Coin;
+import ch.epfl.sdp.item.Healthpack;
+import ch.epfl.sdp.item.ItemBox;
+
+/**
+ *  Control the whole game lifecycle of the solo mode
+ */
+public class Solo implements StartGameController, Updatable {
```

Comment:
Maybe we can extract some methods from this class and the server class (the initializations of the enemies and other things)

Reply:
Yes, I also thought about this before and thinking about the balance between whether to introduce another hierarchy or leave some duplication. Maybe after refactoring the enemyGenerator/itemboxGenerator,  we can make the initializations in the server/solo to have only one or two lines.

Reply:
Yes, we will need to look at it